pkg/static/login.js

```diff
@@ -747,15 +750,48 @@ import "./login.scss";
     function do_hostkey_verification(data) {
         const key_db = get_known_hosts_db();
         const key = data["host-key"];
-        const key_key = key.split(" ")[0];
-        const key_type = key.split(" ")[1];
+        const [key_key, key_type, key_value] = key.split(" ");
+        let db_key = key_db[key_key];
+
+        // did we see this host before?
+        if (db_key) {
+            // exact same key → good
+            if (key_db[key_key] == key) {
+                converse(data.id, data.default);
+                return;
+            }
 
-        if (key_db[key_key] == key) {
-            converse(data.id, data.default);
-            return;
+            /* transition case: cockpit-ssh stored full public keys in the DB, like
+            * "myhost ecdsa-sha2-nistp256 AAAA[..]ToM=\n"; ssh(1) does not show them, only fingerprints, so
+            * cockpit-beiboot only responds with fingerprints as well; they look like "myhost ED25519 SHA256:JVG[..]xs"
+            * check if we can match the given fingerprint to an existing entry */
+            const [db_key_type, db_key_value] = db_key.split(" ").slice(1);
```

Comment:
I guess this could very easily return the wrong number of arguments...

Reply:
Yes it could, hence the code below needs to be careful to get along with an `undefined` db_key_value. That's why it checks for its truthiness first.